@Alyx/getFile.m

```diff
@@ -1,36 +1,49 @@
-function fullPath = getFile(obj, datasetURL)
-%GETFILE Recovers the full filepath of a file on the repository, given the datasetURL
-%	This function is SUPER inefficient because it has to load all
-%	filerecords from the database, and then search for the specific
-%	filerecord whose parent dataset matches the one supplied as input.
+function [fullPath, exists] = getFile(obj, eid, type)
+%GETFILE Recovers the full filepath(s) associated with a given eid
+%	Returns the full path of associated data files given a dataset or file
+%	record URL/eid.  Also returns logical array indicating whether files
+%	exist.
 %
-%   datasetURL: URL of a dataset on Alyx
+%   eid: full URL or eid of a dataset/file record on Alyx, or array thereof
+%   type: string indicating type of eid ('dataset' (default) or 'file')
 %
-% See also ALYX, GETDATA
+% See also ALYX, GETDATA, GETSESSIONS
 %
 % Part of Alyx
-% TODO: Create endpoint so this function is no longer inefficient?
 % 2017 PZH created
+% 2019 MW Rewrote
 
+if nargin < 3; type = 'dataset'; end
 
-% Get all file records
-filerecords = obj.getData('files');
-
-% Extract the datasets which are parent to the filerecords
-datasets = cellfun(@(fr) fr.dataset, filerecords, 'uni', 0);
-
-% Find whichever filerecord has input datasetURL as its parent
-idx = contains(datasets, datasetURL);
-
-if any(idx)
-  fr = filerecords{idx};
-  relPath = fr.relative_path; % Get relative path of file
-  
-  repo = obj.getData('data-repository'); % Get absolute path of repository
-  fullPath = [repo{1}.path relPath]; % Recover the full path of the file on the repository
+% Convert array of strings to cell string
+if isstring(eid) && ~isscalar(eid)
+  eid = cellstr(eid);
 else
-  error('No filerecords with inputted datasetURL as its parent');
+  eid = ensureCell(eid);
 end
 
+% Validate URL
+assert(all(cellfun(@(str)ischar(str) && length(str) >= 36, eid)), 'Invalid eid')
```

Comment:
what is the significance of 36?

Reply:
Alyx uses [UUIDs](https://en.wikipedia.org/wiki/Universally_unique_identifier#Format), which are 36 character strings.  This could be changed to `length(str) == 36` as the length should be constant.

Reply:
got it